cmake/Dependencies.cmake

```diff
@@ -916,17 +916,20 @@ endif()
 
 # ---[ NCCL
 if(USE_NCCL)
-  if(NOT USE_CUDA)
+  if(NOT (USE_CUDA OR USE_ROCM))
     message(WARNING
         "Not using CUDA, so disabling NCCL. Suppress this warning with "
         "-DUSE_NCCL=OFF.")
     caffe2_update_option(USE_NCCL OFF)
   elseif(NOT ${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
```

Comment:
Note to myself: This can be simplified into option dependency. I can make the change after this PR is merged (if you don't wanna do).